api/routes/api_teams.py

```diff
@@ -12,41 +15,50 @@ def handle_exception(e):
     return get_error_response(response, 500)
 
 
+@teams_app.errorhandler(DatabaseConnectionError)
+def handle_db_timeout_error(e):
+    response = {
+        "error": "Database connection timed out. Please try again later.",
+        "details": str(e)
+    }
+    return get_error_response(response, 503)
+
+
 @teams_app.route("/", methods=['GET'])
+@cache.cached(timeout=60*60)
 def get_teams_endpoint():
-    try:
-        all_teams = get_teams(session)
-        return all_teams
-    except Exception as e:
-        print(e)
+    all_teams = get_teams(session)
+    return all_teams
+
+
+@teams_app.route("/<sport_type>", methods=['GET'])
+@cache.cached(timeout=60*60)
+def get_teams_sport_endpoint(sport_type):
+    all_teams = get_teams_sport(session, sport_type)
+    return all_teams
+
+
 @teams_app.route('/statistics', methods=['POST'])
+@cache.cached(timeout=60*1.3)
 def get_teams_statistics_endpoint():
-    try:
-        data = request.get_json()
-        sport = data.get("sport", "Unknown")
-        team_id = data.get("team_id", "Unknown")
-        response = {
-            "sport": sport,
-            "team_id": team_id,
-        }
-        #response = teams_statistics_info()
-        return response
-    except Exception as e:
-        print(e)
+    data = request.get_json()
+    response = {
+        "sport": data.get("sport", "Unknown"),
+        "team_id": data.get("team_id", "Unknown"),
+        "league_id": 3
+    }
+    response = rugby_teams_statistics(response)
```

Comment:
that is now very SOLID, because your first "response" serves as an input to your following method execution.

It'd be better to just do response = rugby_teams_statistics(input_parameters)

I also remind you to use predefined models and DTOs to transfer such data (you can have one for input and different one for output). Everywhere I look I see just plain dictionaries etc etc: it's not convenient and it hinders debug process for whoever decides to do debugging in the future.. Think of a better way to do this

Reply:
I have made UniversalResponseDTO for now
And later when Vlad finishes refactoring all those terrible functions,
I will finish the output for them